prow/cmd/deck/main.go

```diff
@@ -124,6 +138,33 @@ func main() {
 		}
 		mux.Handle("/user-data.js", gziphandler.GzipHandler(handleUserDashboard(ua)))
 	}
+
+	var appClient config.Client
+	if err := yaml.Unmarshal(bytes.TrimSpace(gitAppClientRaw), &appClient); err != nil {
+		logger.WithError(err).Fatal("Error unmarshalling git app client config.")
+	}
+	var cookieSecret config.Cookie
+	if err := yaml.Unmarshal(bytes.TrimSpace(cookieSecretRaw), &cookieSecret); err != nil {
+		logger.WithError(err).Fatal("Error unmarshalling cookie secret.")
+	}
+	decodedSecret, err := hex.DecodeString(cookieSecret.Secret)
+	if err != nil {
+		logger.WithError(err).Fatal("Error decoding cookie secret.")
+	}
+	cookie := sessions.NewCookieStore(decodedSecret)
+
+	gitOAuthConfig := &configAgent.Config().GitOAuthConfig
+	gitOAuthConfig.InitGitOAuthConfig(&appClient, cookie)
+
+	goa := gitoauth.NewGitOAuthAgent(gitOAuthConfig, logrus.WithField("client", "gitoauth"))
+
+	userDashboardAgent := userdashboard.NewDashboardAgent(gitOAuthConfig, logrus.WithField("client", "user-dashboard"))
+	mux.Handle("/user-dashboard", gziphandler.GzipHandler(userDashboardAgent.HandleUserDashboard(userDashboardAgent)))
+	// Handles login request.
+	mux.Handle("/user-dashboard/login", gziphandler.GzipHandler(goa.HandleLogin()))
```

Comment:
feel free to omit gziphandler from these

Reply:
done